src/OpenIddict.Core/OpenIddictExtensions.cs

```diff
@@ -0,0 +1,205 @@
+                    // Copy the OpenIddict options to the ASOS configuration.
+                    options.Options.AuthenticationScheme = instance.AuthenticationScheme;
+
+                    options.Options.Issuer = instance.Issuer;
+
+                    options.Options.AuthorizationEndpointPath = instance.AuthorizationEndpointPath;
+                    options.Options.LogoutEndpointPath = instance.LogoutEndpointPath;
+
+                    options.Options.AccessTokenLifetime = instance.AccessTokenLifetime;
+                    options.Options.AuthorizationCodeLifetime = instance.AuthorizationCodeLifetime;
+                    options.Options.IdentityTokenLifetime = instance.IdentityTokenLifetime;
+                    options.Options.RefreshTokenLifetime = instance.RefreshTokenLifetime;
+
+                    options.Options.ApplicationCanDisplayErrors = instance.ApplicationCanDisplayErrors;
+                    options.Options.AllowInsecureHttp = instance.AllowInsecureHttp;
+                });
+
+                // Register ASP.NET MVC 6 and the actions
+                // associated to the OpenIddict controller.
+                builder.UseMvc(routes => {
+                    // Register the actions corresponding to the authorization endpoint.
+                    if (instance.AuthorizationEndpointPath.HasValue) {
+                        routes.MapRoute("{D97891B4}", instance.AuthorizationEndpointPath.Value.Substring(1), new {
+                            controller = typeof(OpenIddictController<,>).Name,
+                            action = nameof(OpenIddictController<dynamic, dynamic>.Authorize)
+                        });
+
+                        routes.MapRoute("{7148DB83}", instance.AuthorizationEndpointPath.Value.Substring(1) + "/accept", new {
+                            controller = typeof(OpenIddictController<,>).Name,
+                            action = nameof(OpenIddictController<dynamic, dynamic>.Accept)
+                        });
+
+                        routes.MapRoute("{23438BCC}", instance.AuthorizationEndpointPath.Value.Substring(1) + "/deny", new {
+                            controller = typeof(OpenIddictController<,>).Name,
+                            action = nameof(OpenIddictController<dynamic, dynamic>.Deny)
+                        });
+                    }
+
+                    // Register the action corresponding to the logout endpoint.
+                    if (instance.LogoutEndpointPath.HasValue) {
+                        routes.MapRoute("{C7DB102A}", instance.LogoutEndpointPath.Value.Substring(1), new {
+                            controller = typeof(OpenIddictController<,>).Name,
+                            action = nameof(OpenIddictController<dynamic, dynamic>.Logout)
+                        });
+                    }
+                });
+            }, services => {
+                services.AddAuthentication();
+                services.AddCaching();
+                services.AddCors();
```

Comment:
I think AddMvc() already adds Cors and Cache, there's no need to Add them again (?)